File: providers/aws/src/test/java/datagen/services/ec2/model/Ipv6RangeGenerator.java
/*-
 * #%L
 * CloudSpec AWS Provider
 * %%
 * Copyright (C) 2020 Ezequiel Foncubierta
 * %%
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 * 
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 * 
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 * #L%
 */
package datagen.services.ec2.model;

import datagen.BaseGenerator;
import software.amazon.awssdk.services.ec2.model.Ipv6Range;

import java.util.List;

public class Ipv6RangeGenerator extends BaseGenerator {
    public static List<Ipv6Range> ipv6Ranges() {
        return ipv6Ranges(faker.random().nextInt(1, 10));
    }

    public static List<Ipv6Range> ipv6Ranges(Integer n) {
        return listGenerator(n, Ipv6RangeGenerator::ipv6Range);
    }

    public static Ipv6Range ipv6Range() {
        return Ipv6Range.builder()
                        .cidrIpv6(faker.internet().ipV6Cidr())
                        .description(faker.lorem().sentence())
                        .build();
    }
}

File: providers/aws/src/test/kotlin/datagen/services/ec2/model/Ipv6RangeGenerator.kt
/*-
 * #%L
 * CloudSpec AWS Provider
 * %%
 * Copyright (C) 2020 Ezequiel Foncubierta
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
package datagen.services.ec2.model

import datagen.BaseGenerator
import software.amazon.awssdk.services.ec2.model.Ipv6Range

object Ipv6RangeGenerator : BaseGenerator() {
    @JvmOverloads
    fun ipv6Ranges(n: Int? = null): List<Ipv6Range> {
        return listGenerator(n) { ipv6Range() }
    }

    fun ipv6Range(): Ipv6Range {
        return Ipv6Range.builder()
                .cidrIpv6(faker.internet().ipV6Cidr())
                .description(faker.lorem().sentence())
                .build()
    }
}
